### PR TITLE
Refactor/system component

### PR DESCRIPTION
**Pull request by Matthew-Kulich**

### CODE REVIEW

File: src/utils/JsonLdUtils.tsx [hunk 1]
```diff
@@ -43,19 +52,25 @@ export default class JsonLdUtils {
      * array is returned.
      * @param compacted Compacted JSON-LD
      */
-    public static loadArrayFromCompactedGraph<T extends JsonLdDictionary>(compacted: object): T[] {
+    public static loadArrayFromCompactedGraph<T extends JsonLdDictionary>(
+        compacted: object
+    ): T[] {
         if (!compacted.hasOwnProperty("@context")) {
-            return []
+            return [];
```
Comment (kostobog) [inline src/utils/JsonLdUtils.tsx:59]: change

File: src/utils/JsonLdUtils.tsx [hunk 2]
```diff
@@ -43,19 +52,25 @@ export default class JsonLdUtils {
...
         }
-        return compacted.hasOwnProperty("@graph") ? Object.keys(compacted["@graph"]).map(k => compacted["@graph"][k]) : [compacted]
+        return compacted.hasOwnProperty("@graph")
+            ? Object.keys(compacted["@graph"]).map((k) => compacted["@graph"][k])
```
Comment (kostobog) [inline src/utils/JsonLdUtils.tsx:62]: change

File: src/utils/JsonLdUtils.tsx [hunk 3]
```diff
@@ -67,32 +82,39 @@ export default class JsonLdUtils {
...
+                            if (reference) {
+                                val[i] = reference;
+                            } else {
+                                JsonLdUtils.processNode(val[i], idMap);
+                            }
                         }
                     }
+                } else if (typeof val === "object") {
+                    const reference = JsonLdUtils.getReferencedNodeIfExists(val, idMap);
+                    if (reference) {
```
Comment (kostobog) [inline src/utils/JsonLdUtils.tsx:105]: change

**Review by kostobog — COMMENTED** [review]
I find only 3 actual changes the rest formatting changes. Is this ok?

**Comment by Matthew-Kulich** [comment]
> I find only 3 actual changes the rest formatting changes. Is this ok?
Yes, it is ok. I just directly copied it from the updated termit-ui version and I didn't notice that the only actual change  was
else if statement and formatting.

**Review by kostobog — COMMENTED** [review]
Looks good to me.

**Pull request merged**